app/assets/javascripts/discourse/app/components/modal/second-factor-add-totp.js

```diff
@@ -0,0 +1,66 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import { inject as service } from "@ember/service";
+import I18n from "I18n";
+
+export default class SecondFactorAddTotp extends Component {
+  @service modal;
+
+  @tracked loading = false;
+  @tracked secondFactorImage;
+  @tracked secondFactorKey;
+  @tracked showSecondFactorKey = false;
+  @tracked errorMessage;
+  @tracked secondFactorToken;
+
+  @action
+  totpRequested() {
+    this.args.model.secondFactor
+      .createSecondFactorTotp()
+      .then((response) => {
+        if (response.error) {
+          this.errorMessage = response.error;
+          return;
+        }
+
+        this.errorMessage = null;
+        this.secondFactorKey = response.key;
+        this.secondFactorImage = response.qr;
+      })
+      .catch((error) => {
+        this.modal.close();
```

Comment:
```suggestion
        this.args.closeModal();
```
(and drop the service injection)